src/operator/effectivepolicy/groupreconciler.go

```diff
@@ -95,35 +97,59 @@ func (g *GroupReconciler) getAllServiceEffectivePolicies(ctx context.Context) ([
 		if err != nil {
 			return nil, err
 		}
-		if intent, ok := serviceToIntent[service]; ok {
-			ep.ClientIntent = lo.ToPtr(intent)
+		// Ignore intents in deletion process
+		if clientIntents, ok := serviceToIntent[service]; ok && clientIntents.DeletionTimestamp.IsZero() && clientIntents.Spec != nil {
+			ep.Calls = append(ep.Calls, clientIntents.GetCallsList()...)
+			ep.ClientIntentsEventRecorder = injectablerecorder.NewObjectEventRecorder(&g.InjectableRecorder, lo.ToPtr(clientIntents))
 		}
 		epSlice = append(epSlice, ep)
 	}
 
 	return epSlice, nil
 }
 
+// a function that checks if we should create a SEP for a given intent target server
```

Comment:
Is the comment necessary? Seems to be saying the same thing as the function's name and is likely to go out of date when refactored :)
If you absolutely need to keep it, consider:
```suggestion
// shouldCreateEffectivePolicyForIntentTargetServer that checks if we should create a SEP for a given intent target server
```
So that IDEs will find this line as well and hopefully the person refactoring will notice they need to update the comment.